### PR TITLE
RasterDataset TS support

### PR DESCRIPTION
**Pull request by sfalkena**
Very basic ts support without any implementation currently in the datasets. In VERS repo, this is supported, but implementation in RasterDataset and others is still to come.

The reason for this separate PR is so that we can move on within VERS, while developing full TS in sync with TG in parallel

### CODE REVIEW

**Review by am94ghiassi — APPROVED** [review]

File: torchgeo/datasets/geo.py [hunk 1]
```diff
@@ -510,6 +519,26 @@ def __init__(
...
+        Args:
+            filepaths (list): A list of filepaths.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the extracted file metadata.
+        """
+        filename_regex = re.compile(self.filename_regex, re.VERBOSE)
+        file_metadata = []
+        for filepath in filepaths:
+            match = re.match(filename_regex, filepath)
```
Comment (am94ghiassi) [inline torchgeo/datasets/geo.py:534]: suggestion: `match = filename_regex.match(filepath)`

File: torchgeo/samplers/single.py [hunk 2]
```diff
@@ -145,11 +145,8 @@ def __iter__(self) -> Iterator[BoundingBox]:
             if self.dataset.return_as_ts:
                 mint = self.dataset.bounds.mint
```
Comment (am94ghiassi) [inline torchgeo/samplers/single.py]: Notebook fails here

File: torchgeo/datasets/geo.py (hunk 1 above)
Comment (sfalkena) [reply torchgeo/datasets/geo.py:534]: This is the method used in TorchGeo, so I try to stick to that

File: torchgeo/samplers/single.py (hunk 2 above)
Comment (sfalkena) [reply torchgeo/samplers/single.py]: After merging in main, it no longer complains

**Pull request merged**